index.ts

```diff
@@ -0,0 +1,139 @@
+import express from "express";
+import {exec} from "child_process";
+import crypto from "crypto";
+import dotenv from "dotenv";
+
+dotenv.config();
+
+const app = express();
+const port = process.env.PORT || '3333';
+const secret = process.env.GITHUB_SECRET;
+if(!secret) {
+  console.error("Error: GITHUB_SECRET environment variable is not set.");
+  process.exit(1);
+};
+
+let isDocumentationWebsiteUpdated = false;
+let isMindmapUpdated = false;
+let contributorsBuildRequired = false;
+
+let documentationWebsiteBuildTime = 0;
+let mindmapBuildTime = 0;
+let contributorsBuildTime = 0;
+
+app.use(express.json());
+
+app.post("/webhook", async (req, res) => {
+  console.log("req receieved");
+  const signature = req.headers["x-hub-signature"] as string;
+
+  const payload = JSON.stringify(req.body);
+  const hmac = crypto.createHmac("sha1", secret);
+
+  const calculatedSignature = `sha1=${hmac.update(payload).digest("hex")}`;
+  console.log("cals", calculatedSignature) // need this to stay as number seems to change
+
+  if (crypto.timingSafeEqual(Buffer.from(signature), Buffer.from(calculatedSignature))) {
+    const {result, respMessage} = await getBranchStatus(req.body);
+    console.log("Result: ", result);
+    res.status(200).send({ result, respMessage });
+  } else {
+    res.status(400).send("Invalid Github signature");
+  }
+});
+
+interface BranchStatus {
+  result:  number | string;
+  respMessage: string ;
+};
+
+app.listen(port, () => {
+  console.log(`Server listening on port ${port}`);
+});
+
+const sanitize = (cmd: string) => {
+  const sanitized = cmd.replace(/[;&|`<>$]/g, '');
+
+  if(/[\r\n\t]/.test(cmd)) {
+    throw new Error("Invalid characters in command");
+  }
+
+  return sanitized;
+};
+
+const executeCmd = async (cmd: string) => {
+  try {
+    const {stdout, stderr} = await exec(sanitize(cmd));
+    return stderr + "\n" + stdout;
+  } catch (error: any) {
```

Comment:
no any under any circumstances, this is likely an Error or at least unknown

Reply:
removed any.